packages/flutter/lib/src/gestures/pointer_router.dart

```diff
@@ -94,19 +94,23 @@ class PointerRouter {
   /// Routes are called in the order in which they were added to the
   /// PointerRouter object.
   void route(PointerEvent event) {
-    final LinkedHashSet<_RouteEntry> routes = _routeMap[event.pointer];
-    final List<_RouteEntry> globalRoutes = List<_RouteEntry>.from(_globalRoutes);
+    final LinkedHashMap<PointerRoute, Matrix4> routes = _routeMap[event.pointer];
+    final LinkedHashMap<PointerRoute, Matrix4> copiedGlobalRoutes = LinkedHashMap<PointerRoute, Matrix4>.from(_globalRoutes);
     if (routes != null) {
-      for (_RouteEntry entry in List<_RouteEntry>.from(routes)) {
-        if (routes.any(_RouteEntry.isRoutePredicate(entry.route)))
-          _dispatch(event, entry);
-      }
-    }
-    for (_RouteEntry entry in globalRoutes) {
-      if (_globalRoutes.any(_RouteEntry.isRoutePredicate(entry.route)))
-        _dispatch(event, entry);
+      _dispatchEventToRoutes(event, routes, LinkedHashMap<PointerRoute, Matrix4>.from(routes));
     }
+    _dispatchEventToRoutes(event, _globalRoutes, copiedGlobalRoutes);
   }
+
+  void _dispatchEventToRoutes(
+    PointerEvent event,
+    LinkedHashMap<PointerRoute, Matrix4> referenceRoutes,
+    LinkedHashMap<PointerRoute, Matrix4> copiedRoutes,
+    ) => copiedRoutes.forEach((PointerRoute route, Matrix4 transform) {
```

Comment:
formatting nit: the ) should be aligned with the void.

Also: Flutter Style only allows the => for functions that fit in one line. This one doesn't, so please change this to use { } style.

Reply:
Fixed. I've also broken some lines for fitting them in 80 chars in this [commit](https://github.com/flutter/flutter/pull/42526/commits/d5d3bf3542530afc3a0500db0b1c185b365a69ef). Is this OK? Or is it better to revert these changes?